### DISCUSSION TITLE
[@types/react] React.memo and generics won't work

### DISCUSSION THREAD

**Discussion by tal-rofe** [DefinitelyTyped/DefinitelyTyped#62602] [Issues with a @types package]
When using `React.memo` and generics in Typescript, there is an issue with props, at-least for me, when providing a function in props.

React version: 18.2.0

## Steps To Reproduce

1. Typescript will complain for props types immediately when compiling Typescript

Link to code example:
https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4Aoc4AOxiSk3STgEkB5MGYCagHgBUAfHADe5OBLhEUAEx4AbAJ5wAbinkBXJAC44-CpKmo51JXHkoARknm6AzjCg0A5hQC+lGnQZNWABRwwOwFhMUNpEzMuGHkdOAcnaldxSQiFZQhObmo7XXYsnhCAbQBdA1TjdPibJAwkGQ4uHl19FIk00wzqAGUajEbs3QAKNU04-gBKOABeYRUIYBl3SjQeBzgAMWB5bwA1YCQAdxm4AQAaQSGwQNzEYhgAOgDMuwB1YBgACwBhT+2ZIh8FjPIIhQRTWaiNpGGAaKDUOBDaGGXgyYAqQTIwySXifACsghE1xeD2isTcvAA9PjMVjsaJiUEHpkmjkHiAUGAhkMWdkznAaDIkAAPCHCJH0yU4ywaGAwHh0qXYgDWSEU0xEgpFHiVurgMEUYCQ0wARDK5TwTYq9XAeN95MA0MqNUMxXBGXZmT0+jABjweQVqA9RloJjqbZJMRHJSJeTwHhZrPJw9Gqeb5dQo3qJmHFVS0RjoRN3BQVmt4FsdvR9kcAKLCyCweonZDoR4gJAgCBDSt7A6HYueWj0RhoZjAm4AJhCUPClU6+o+sXsjhc5Xa87McZyeT9fCEpXXRlkVTsPvqe5aR46W+9sX6geGIfGbvmi2W5FWOQr228J3OlwerorYYE8NzvF8vz-ICvATi805COCMzitCvC9tW-b1o2dAyIqZLGkSNykkuSApti252BqHrMoGdhkYYZ73jhe5UURjG1Mxgb0ZIPC9ExLGESSvE+nubhwJSlJwGwyooIo0KUpig5AA

## The current behavior
Typescript complain about function in props

## The expected behavior
Should not complain

## Manual fix
Can be fixed if doing:
```ts
declare module "react" { // augment React types
    function memo<A, B>(Component: (props: A) => B): (props: A) => React.ReactElement | null
    // return type is same as ReturnType<ExoticComponent<any>>
}
```

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
Haven't thought about your suggestion. Seems to work at a glance 👍🏻  The tricky part might be to get this in without breaking existing code. Would you be interested in preparing a PR so that we can see the impact? Ideally with `A` and `B` being constrained to something useful e.g. `A` should probably extend `object` and `B` a valid return value of components (according to TypeScript).

**Reply by tal-rofe** [reply]
I'm sorry, I wish I had to time to make a PR.
Please keep this discussion open so others can relate it either with a PR or an other issue/discussion.

Thanks
